src/compute/src/render/context.rs

```diff
@@ -1073,18 +981,8 @@ where
         key: &Vec<MirScalarExpr>,
         thinning: &Vec<usize>,
         key_val_types: Option<(Vec<ColumnType>, Vec<ColumnType>)>,
-    ) -> (SpecializedArrangement<S>, Collection<S, DataflowError, i64>) {
-        if let Some((_key_types, val_types)) = key_val_types {
-            if val_types.is_empty() {
-                // Emtpy value specialization.
-                let (oks, errs) = oks.map_fallible(
-                    "FormArrangementKey [val: empty]",
-                    specialized_arrangement_key(key.clone(), thinning.clone(), None, Some(vec![])),
-                );
-                let name = &format!("{} [val: empty]", name);
-                let oks = oks.mz_arrange::<RowSpine<_, _>>(name);
-                return (SpecializedArrangement::RowUnit(oks), errs);
-            }
+    ) -> (MzArrangement<S>, Collection<S, DataflowError, i64>) {
+        if let Some((_key_types, _val_types)) = key_val_types {
         } else {
             // We expect to always have type information for arrangements, do getting here is
             // likely a bug.
```

Comment:
The containing function `specialized_arrange` should probably go away, too.

Reply:
I think that makes sense, but tbh I didn't get as far as "re-organize the code" beyond removing the various `dispatch_` methods. This seemed larger than we'd want to inline, or at least than I was certain we wanted to inline. No strong opinions on my part, other than that I was not certain enough myself.